pkg/k8s/watchers/pod.go

```diff
@@ -244,6 +249,12 @@ func (k *K8sWatcher) updateK8sPodV1(oldK8sPod, newK8sPod *slim_corev1.Pod) error
 		if err != nil {
 			return err
 		}
+
+		err = updateCiliumEndpointLabels(podEP, newPodLabels)
+		if err != nil {
+			log.WithError(err).Debugf("Error while updating CiliumEndpoint object with new Pod labels")
+			return err
+		}
```

Comment:
This should be replaced with a controller so that it runs in the background. This controller needs to be associated with the endpoint `podEP` so that the controller is deleted once the EP is gone.